### ISSUE TITLE
Can we use it to extract features from JPG and PNG images?

### ISSUE THREAD

**Issue by yuehailin** [Radiomics/pyradiomics#576]
A clear and concise question regarding your extraction. Please check [FAQ section]
My image types are JPG and PNG. How can I use pyradiomics to extract the omics features?

**PyRadiomics configuration**
File path configuration file
ID,Image,Mask
brain1,../data/0.jpg,../data/0-mask.jpg

The yaml file
imageType:
  Original: {}

featureClass:
  firstorder: null

  shape: null
setting:
  binWidth: 25
  label: 1
  padDistance: 10
  resampledPixelSpacing:
  - 1
  - 1
  - 1
  voxelArrayShift: 100

error：
RuntimeError: Exception thrown in SimpleITK LabelStatisticsImageFilter_Execute: d:\a\1\sitk\code\common\include\sitkDualMemberFunctionFactory.hxx:214:
sitk::ERROR: Pixel type: vector of 8-bit unsigned integer is not supported in 2D byclass itk::simple::LabelStatisticsImageFilter

I used batchprocessing.py in the examples for feature extraction

 - OS: windows 10
 - Python version: python 3.7
 - PyRadiomics version ：2.2.0

You can give some advice or guidance

Thank you very much！

**Comment by ayou-you** [comment]
i have the same question still not be solved yet, but i found an alternative method. i read the png by sitk.ReadImage, and then use sitk.WriteImage to save the png as nrrd, so that it could be read by pyradiomics.
the code i use:
image = sitk.ReadImage(image_path) 
sitk.WriteImage(image, image_path.split(".png")[0] + ".nrrd")

**Comment by panos1998** [comment]
> i have the same question still not be solved yet, but i found an alternative method. i read the png by sitk.ReadImage, and then use sitk.WriteImage to save the png as nrrd, so that it could be read by pyradiomics. the code i use: image = sitk.ReadImage(image_path) sitk.WriteImage(image, image_path.split(".png")[0] + ".nrrd")

Radiomics needs additional info such as origin, direction matrix and pixel spacing

**Comment by JoostJM** [comment]
Hello,

This is covered in the online documentation.
Tldr: pyradiomics requires grayscale images as input (i.e. single values per pixel). Jpg and png are color images, with 3-4 values per pixel. You need to extract the color channel you want to extract from.

Origin and spacing are not required, but do provide meaningful extra information. Specifically, shape features may be meaningless.
